src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Commands/ObjectBrowserCommand.cs

```diff
@@ -0,0 +1,142 @@
+/// </item>
+/// <item>
+/// Relatable items (attached sub-nodes) modelled as <see cref="IRelatableItem"/>, must implement <see cref="IObjectBrowserItem"/>.
+/// </item>
+/// </list>
+/// </remarks>
+[ProjectCommand(VSConstants.CMDSETID.StandardCommandSet2K_string, (long)VSConstants.VSStd2KCmdID.VIEWREFINOBJECTBROWSER)]
+[AppliesTo(ProjectCapability.DependenciesTree)]
+[Order(Order.Default)]
+[method: ImportingConstructor]
+internal sealed class ObjectBrowserCommand(UnconfiguredProject project, IVsUIService<SVsObjBrowser, IVsNavigationTool> navigationTool) : AbstractSingleNodeProjectCommand
+{
+    // This code needs a little explanation. Consider a tree like this:
+    //
+    // - Dependencies
+    //   - Packages
+    //     - MyPackage
+    //       - MyPackage.dll
+    // - Frameworks
+    //   - Microsoft.NETCore.App
+    //     - Microsoft.CSharp
+    //
+    // The "MyPackage" and "Microsoft.NETCore.App" nodes are top-level dependencies, and they have IProjectTree nodes.
+    //
+    // The "MyPackage.dll" and "Microsoft.CSharp" nodes are transitive nodes, added via the "AttachedCollections" API
+    // (IRelatableItem) and do not have IProjectTree nodes.
+    //
+    // When the user right-clicks on an IRelatableItem node, CPS is called with the IProjectTree node of the closest
+    // ancestor, which is not the node the user clicked on. There's no easy way to thread that state through on the
+    // call stack, so we use a static collection on the menu controller that's only populated while the context menu
+    // is open. If it has items here, we know we're in the transitive case and will attempt to handle the command.
+
+    protected override async Task<CommandStatusResult> GetCommandStatusAsync(IProjectTree node, bool focused, string? commandText, CommandStatus progressiveStatus)
+    {
+        string? path = await TryGetAssemblyPathAsync(node);
+        
+        if (path is not null)
+        {
+            // We handle this .
+            return new CommandStatusResult(handled: true, commandText, CommandStatus.Enabled | CommandStatus.Supported);
+        }
+
+        // We won't handle this. Allow other handlers to do so.
+        return CommandStatusResult.Unhandled;
+    }
+
+    protected override async Task<bool> TryHandleCommandAsync(IProjectTree node, bool focused, long commandExecuteOptions, IntPtr variantArgIn, IntPtr variantArgOut)
+    {
+        string? path = await TryGetAssemblyPathAsync(node);
+
+        if (path is not null)
+        {
+            ShowAssemblyInObjectBrowser();
+
+            // Handled.
+            return true;
+        }
+
+        // Unhandled. Should not happen in the context menu, as we only show the command when we can handle it.
+        // Might theoretically happen if someone invoked the command directly on the tree.
+        return false;
+
+        void ShowAssemblyInObjectBrowser()
+        {
+            // We're opening a file as a container (DLL).
```

Comment:
just curious, what does it mean to open a file as a container specifically? what other ways are supported?

Reply:
"Container" is terminology from the Object Browser APIs. It basically means DLL, in that it contains types etc. You can also open at a specific symbol (method, property, type, etc), but we just want the DLL.